src/main/java/com/webdev/cosmo/cosmobackend/service/common/mapper/UserMapper.java

```diff
@@ -0,0 +1,22 @@
+package com.webdev.cosmo.cosmobackend.service.common.mapper;
+
+import com.webdev.cosmo.cosmobackend.service.api.ROLE;
+import com.webdev.cosmo.cosmobackend.service.api.User;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+import org.openapitools.model.UserModel;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Mapper
+@Component
+public abstract class UserMapper {
```

Comment:
i think You can declare it as interface, without @Component. There is already predefined in pom.xml configuration sping model compoonent for mapstruct